pennylane/queuing.py

```diff
@@ -359,8 +359,9 @@ def append(self, obj, **kwargs):
         self[obj] = kwargs
 
     def remove(self, obj):
-        """Remove ``obj`` from the queue.  Raises ``KeyError`` if ``obj`` is not already in the queue."""
-        del self[obj]
+        """Remove ``obj`` from the queue. Passes silently if the object is not in the queue."""
+        if obj in self:
+            del self[obj]
 
     def update_info(self, obj, **kwargs):
         """Update ``obj``'s metadata with ``kwargs`` if it exists in the queue."""
```

Comment:
Should we remove/deprecate `update_info`?

Reply:
If we remove it, I believe we should also remove it from `docs/code/api`

Reply:
What about deprecating it? If we don't do anything we might end up forgetting we need to remove it

Reply:
Maybe in a later PR.